### PR TITLE
Use TestTable in faker

### PR DESCRIPTION
**Pull request by wendigo**
## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

**Review by nineinchnick — APPROVED** [review]

File: plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java [hunk 1]
```diff
@@ -42,26 +43,23 @@ void testShowTables()
...
     {
-        assertUpdate("CREATE TABLE faker.default.comments (id INTEGER, name VARCHAR)");
-        assertUpdate("COMMENT ON COLUMN comments.name IS 'comment text'");
-
-        assertTableColumnNames("faker.default.comments", "id", "name");
-
-        assertUpdate("DROP TABLE faker.default.comments");
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "cannot_comment", "(id INTEGER, name VARCHAR)")) {
+            assertUpdate("COMMENT ON COLUMN %s.name IS 'comment text'".formatted(table.getName()));
+            assertTableColumnNames("faker.default." + table.getName(), "id", "name");
```
Comment (ebyhr) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:48]: We should verify column comments, not name.

Comment (ebyhr) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:48]: `faker.default.` is redundant.

File: plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java [hunk 2]
```diff
@@ -42,26 +43,23 @@ void testShowTables()
...
-                .hasMessageContaining("Cannot set comment for $row_id column");
-
-        assertUpdate("DROP TABLE faker.default.cannot_comment");
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "cannot_comment", "(id INTEGER, name VARCHAR)")) {
+            @Language("SQL")
+            String testQuery = "COMMENT ON COLUMN \"%s\".\"$row_id\" IS 'comment text'".formatted(table.getName());
+
+            assertThatThrownBy(() -> getQueryRunner().execute(testQuery))
+                    .as("Query: " + testQuery)
+                    .hasMessageContaining("Cannot set comment for $row_id column");
```
Comment (ebyhr) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:61]: Use `assertThat(query())` instead and inline `testQuery`. (No need to use it in `as()`)

File: plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java [hunk 3]
```diff
@@ -212,90 +210,71 @@ rnd_nchar char(1000) NOT NULL,
...
-        assertUpdate(tableQuery);
-
-        @Language("SQL")
-        String testQuery = "SELECT count(distinct rnd_bigint) FROM default_table_limit";
-        assertQuery(testQuery, "VALUES (100)");
-
-        assertUpdate("DROP TABLE faker.default.default_table_limit");
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "default_table_limit", "(rnd_bigint bigint NOT NULL) WITH (default_limit = 100)")) {
+            @Language("SQL")
+            String testQuery = "SELECT count(distinct rnd_bigint) FROM " + table.getName();
```
Comment (ebyhr) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:252]: Inline `testQuery`.

File: plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java [hunk 4]
```diff
@@ -212,90 +210,71 @@ rnd_nchar char(1000) NOT NULL,
...
-
-        assertUpdate("DROP TABLE faker.default.generators");
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "generators", """
+            (
+                name VARCHAR NOT NULL WITH (generator = '#{Name.first_name} #{Name.last_name}'),
+                age_years INTEGER NOT NULL
+            )
+            """)) {
+            @Language("SQL")
+            String testQuery = "SELECT count(name) FILTER (WHERE LENGTH(name) - LENGTH(REPLACE(name, ' ', '')) = 1) FROM " + table.getName();
```
Comment (ebyhr) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:275]: Inline `testQuery`.

**Review by ebyhr — APPROVED** [review]

File: plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java [hunk 5]
```diff
@@ -42,26 +43,22 @@ void testShowTables()
...
     {
-        assertUpdate("CREATE TABLE faker.default.cannot_comment (id INTEGER, name VARCHAR)");
-        @Language("SQL")
-        String testQuery = "COMMENT ON COLUMN \"cannot_comment\".\"$row_id\" IS 'comment text'";
-
-        assertThatThrownBy(() -> getQueryRunner().execute(testQuery))
-                .as("Query: " + testQuery)
-                .hasMessageContaining("Cannot set comment for $row_id column");
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "cannot_comment", "(id INTEGER, name VARCHAR)")) {
+            String testQuery = "COMMENT ON COLUMN \"%s\".\"$row_id\" IS 'comment text'".formatted(table.getName());
```
Comment (ebyhr) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:56]: Inline `testQuery`.

**Review by ebyhr — APPROVED** [review]

File: plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java [hunk 6]
```diff
@@ -42,26 +43,22 @@ void testShowTables()
...
 
-        assertUpdate("DROP TABLE faker.default.cannot_comment");
+            assertThat(query(testQuery))
+                    .nonTrinoExceptionFailure()
+                    .hasMessageContaining("Cannot set comment for $row_id column");
```
Comment (ebyhr) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:60]: https://github.com/trinodb/trino/blob/115b8ea7a375ba820a5ce98688594fa9f8c85489/plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerMetadata.java#L265-L267 should throw `TrinoException`.

Comment (wendigo) [reply plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:60]: Done

File: plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java [hunk 7]
```diff
@@ -42,26 +44,21 @@ void testShowTables()
...
     {
-        assertUpdate("CREATE TABLE faker.default.comments (id INTEGER, name VARCHAR)");
-        assertUpdate("COMMENT ON COLUMN comments.name IS 'comment text'");
-
-        assertTableColumnNames("faker.default.comments", "id", "name");
-
-        assertUpdate("DROP TABLE faker.default.comments");
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "comment", "(id INTEGER, name VARCHAR)")) {
+            assertUpdate("COMMENT ON COLUMN %s.name IS 'comment text'".formatted(table.getName()));
+            assertQuery("SHOW COLUMNS FROM " + table.getName(), "VALUES ('id', 'integer', '', ''), ('name', 'varchar', '', 'comment text')");
```
Comment (brybacki) [inline plugin/trino-faker/src/test/java/io/trino/plugin/faker/TestFakerQueries.java:49]: 👍

**Pull request merged**